src/topology/algebra/order/lower.lean

```diff
@@ -0,0 +1,240 @@
+
+## Tags
+
+lower topology, preorder
+-/
+
+universe u
+variable {α : Type u}
+
+open set topological_space
+
+/--
+Type synonym for a preorder equipped with the lower topology
+-/
+@[derive preorder, nolint unused_arguments]
+def lower (α : Type u) [preorder α] := α
+
+instance (α : Type u) [preorder α] [p : nonempty α] : nonempty (lower α) := p
+
+instance (α : Type u) [preorder α] [p : inhabited α] : inhabited (lower α) := p
+
+/--
+The lower topology is the topology generated by the complements of the closed intervals to infinity
+-/
+def lower_topology (α : Type u) [preorder α] : topological_space α :=
+generate_from {s | ∃ a, (Ici a)ᶜ = s}
+
+instance lower.topological_space (α : Type u) [preorder α] :
+  topological_space (lower α) := lower_topology (lower α)
+
+
+section pre_order
+
+
+variable [preorder α]
+
+lemma is_open_iff_generate_Ici_comp {s : set (lower α)} :
+  is_open s ↔ generate_open {s | ∃ a, (Ici a)ᶜ = s} s := iff.rfl
+
+/-
+Left-closed right-infinite intervals [a,∞) are closed in the lower topology
+-/
+lemma is_closed_Ici (a : lower α) : is_closed (Ici a) :=
+is_open_compl_iff.1 $ generate_open.basic _ ⟨a, rfl⟩
+
+/-
+The upper closure of a finite subset is closed in the lower topology
+-/
+lemma upper_closure_is_closed (F : set (lower α)) (h : F.finite) :
+  is_closed (upper_closure F : set (lower α)) :=
+begin
+  simp only [← upper_set.infi_Ici, upper_set.coe_infi],
+  exact is_closed_bUnion h (λ a h₁, is_closed_Ici a),
+end
+
+/-
+Every subset open in the lower topology is a lower set
+-/
+lemma lower_open_is_lower {s : set (lower α)} (h: is_open s) : is_lower_set s :=
+begin
+  rw is_open_iff_generate_Ici_comp at h,
+  induction h,
+  case topological_space.generate_open.basic : u
+  { rw mem_set_of_eq at h_H,
+    choose a h_H using h_H,
+    rw ← h_H,
+    apply is_upper_set.compl,
+    apply is_upper_set_Ici, },
+  case topological_space.generate_open.univ : { exact is_lower_set_univ },
+  case topological_space.generate_open.inter : u v hu1 hv1 hu2 hv2
+    { apply is_lower_set.inter hu2 hv2 },
+  case topological_space.generate_open.sUnion : { apply is_lower_set_sUnion h_ih, },
+end
+
+/-
+The closure of a singleton {a} in the lower topology is the left-closed right-infinite interval
+[a,∞)
+-/
+lemma lower_topology.closure_singleton (a : lower α) : closure {a} = Ici a :=
+begin
+  rw subset_antisymm_iff,
+  split,
+  { apply closure_minimal _ (is_closed_Ici a), rw [singleton_subset_iff, mem_Ici], },
+  { unfold closure,
+    refine subset_sInter _,
+    intro u,
+    intro h,
+    rw mem_set_of_eq at h,
+    intro b,
+    intro hb,
+    rw mem_Ici at hb,
+    rw [singleton_subset_iff, ← is_open_compl_iff] at h,
+    by_contradiction H,
+    rw ← mem_compl_iff at H,
+    have h1: a ∈ uᶜ, from lower_open_is_lower h.left hb H,
+    rw mem_compl_iff at h1,
+    rw ← not_not_mem at h,
+    apply absurd h1 h.right, },
+end
+
+end pre_order
+
+section partial_order
+
+variable [partial_order α]
```

Comment:
This is only necessary for the t0_space instance. You could extend the `pre_order` section all the way down there.
```suggestion
```